nx/lib/nx/defn/grad.ex

```diff
@@ -440,6 +440,77 @@ defmodule Nx.Defn.Grad do
     grad_pairs(pairs, g, cache)
   end
 
+  defp grad(:sort, [tensor, opts, _], ans, g, cache) do
+    # shape = tensor[0].shape
+    # iotas = []
+    # for dim, size in enumerate(shape):
+    #   dtype = np.int32 if size < np.iinfo(np.int32).max else np.int64
+    #   iotas.append(broadcasted_iota(dtype, shape, dim))
+    # sorted_tensor = sort_p.bind(*(tensor + (iotas[dimension],)), dimension=dimension,
+    #                       is_stable=is_stable, num_keys=num_keys)
+    # idx = tuple(sorted_tensor[-1] if i == dimension else iotas[i]
+    #             for i in range(len(shape)))
+    # g_out = tuple(t if type(t) is ad_util.Zero else t[idx] for t in g)
+    # return tuple(sorted_tensor[:-1]), g_out
```

Comment:
This is the code I tried to translate from Jax.
It seems that the basic idea is to retrieve the corresponding grads for each "line" of the sorted tensor, by logical indexing.

However, we don't have logical indexing available, so I tried another approach by concatenating a sequence of numbers to the tensor, slicing it afterwards. However, I couldn't find a way to convert the resulting {n} tensor into a list of integers, which in turn makes the construction of `starts` impossible.

Any tips on how we could approach this?

Reply:
What is logical indexing in this case? Are you selecting only some lines and ignoring the other ones? Or are you re-ordering?

Reply:
From what I gathered from the code, it basically reorders the gradient tensor according to the ordering applied to the original tensor